awx/api/serializers.py

```diff
@@ -5490,6 +5490,12 @@ def get_field_from_model_or_attrs(fd):
 
         if peers_from_control_nodes and node_type not in (Instance.Types.EXECUTION, Instance.Types.HOP):
             raise serializers.ValidationError("peers_from_control_nodes can only be enabled for execution or hop nodes.")
+
+        if self.instance.node_type == 'control' and self.instance.peers:
```

Comment:
`self.instance.peers` _is_ valid and expected for control nodes. For example, add an execution node via the API with `peers_from_control_nodes` set to True. Now go back and look at the control instance and you'll see `peers` has something in it. If you were to then patch the control instance (say, by changing the capacity_adjustment value), then you'd hit this validation error, which is not what we want.